db/import_listener.go

```diff
@@ -245,7 +246,9 @@ func (il *importListener) ImportFeedEvent(ctx context.Context, collection *Datab
 		// we have attachments to migrate
 		err := collection.MigrateAttachmentMetadata(ctx, docID, event.Cas, syncData)
 		if err != nil {
-			base.WarnfCtx(ctx, "error migrating attachment metadata from sync data to global sync for doc %s. Error: %v", base.UD(docID), err)
+			if !errors.Is(err, gocb.ErrCasMismatch) {
+				base.WarnfCtx(ctx, "error migrating attachment metadata from sync data to global sync for doc %s. Error: %v", base.UD(docID), err)
+			}
 		}
```

Comment:
I'd change this to:

```suggestion
if err != nil && !IsCasMismatch(err, gocb.ErrCasMismatch) {
				base.WarnfCtx(ctx, "error migrating attachment metadata from sync data to global sync for doc %s. Error: %v", base.UD(docID), err)
}
```

The error `gocb.ErrCasMismatch` will only detect the error if it comes from gocb, but if we are running under rosmar the error will be `sgbucket.CasMismatchErr`.

I did double check that `gocb.ErrCasMismatch` is caught by `UpdateXattrs`, and it is due to https://github.com/couchbase/gocb/blob/4d5516ec54c32a09355c42a44e7776cfce236966/kvprovider_coresubdoc.go#L440